### ISSUE TITLE
Panic: attempt to shift left with overflow

### ISSUE THREAD

**Issue by jethrogb** [porglezomp/bitmatch#6]
```rust
#[bitmatch]
let "rxxxxxxx_xxxxxxxx_xxxxxxxx_cccccccc_dddddddd_dddddddd_dddddddd_dddddddd" = u64::from_le_bytes(buf);
```
triggers:
```
error: custom attribute panicked
  --> src/main.rs:34:5
   |
34 |     #[bitmatch]
   |     ^^^^^^^^^^^
   |
   = help: message: attempt to shift left with overflow
```

I'm fairly certain the pattern is correct?

**Comment by porglezomp** [comment]
Ah, I've figured out the problem. The mechanism that generates the bit-masks works via a shift, and it was assuming that it should be doing u32 shifts so generating the `dddd...` mask overflowed the u32 in the process. I'm changing it to generate the masks without needing a specific type, so any legal literal should end up safe.

**Comment by porglezomp** [comment]
I just published v0.1.1 with the fix.
